knack/output.py

```diff
@@ -162,9 +161,9 @@ def out(self, obj, formatter=None, out_file=None):  # pylint: disable=no-self-us
 
     def get_formatter(self, format_type):  # pylint: disable=no-self-use
         # remove color if stdout is not a tty
-        if not sys.stdout.isatty() and format_type == 'jsonc':
+        if not self.cli_ctx.enable_color and format_type == 'jsonc':
```

Comment:
Correct the logic to use `cli_ctx.enable_color` instead of `sys.stdout` to make the decision about falling back from `jsonc` to `json`.